### PR TITLE
[WIP] - Recharts issue 3113 filterProps

### PR DESCRIPTION
**Pull request by BuddhikaGeasman**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/util/svgUtils/types.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+export type SVGElementAttributeTypes = {
+  '*': ReadonlyArray<string>;
```
Comment (ckifer) [inline src/util/svgUtils/types.ts:2]: what is `*`

File: src/util/svgUtils/types.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+export type SVGElementAttributeTypes = {
```
Comment (ckifer) [inline src/util/svgUtils/types.ts:1]: This list should directly reflect https://developer.mozilla.org/en-US/docs/Web/SVG/Element. If there are things that are no longer on that list we can probably clean the list up a bit

File: src/util/ReactUtils.ts [hunk 3]
```diff
@@ -282,18 +207,22 @@ export const isValidSpreadableProp = (
...
-  svgElementType?: FilteredSvgElementType,
+  svgElementType?: SVGElementType,
 ) => {
   /**
    * If the svg element type is explicitly included, check against the filtered element key map
    * to determine if there are attributes that should only exist on that element type.
    * @todo Add an internal cjs version of https://github.com/wooorm/svg-element-attributes for full coverage.
    */
-  const matchingElementTypeKeys = FilteredElementKeyMap?.[svgElementType] ?? [];
+  const svgElementAttributesInCamelCase = getSVGElementAttributesInCamelCase(svgElementAttributes);
```
Comment (ckifer) [inline src/util/ReactUtils.ts:217]: the props we accept are also in camel case for the same reason (react), why do we need to do any transformation here?

File: src/util/ReactUtils.ts [hunk 4]
```diff
@@ -282,18 +207,22 @@ export const isValidSpreadableProp = (
...
+  const matchingElementTypeKeys = svgElementAttributesInCamelCase?.[svgElementType] ?? [];
+
+  const { polygon, polyline, svg, ...rest } = svgElementAttributesInCamelCase;
```
Comment (ckifer) [inline src/util/ReactUtils.ts:220]: why do these three need to be extracted out?

**Review by ckifer — COMMENTED** [review]
Some questions. I don't think we need to do any camel case transformation...

File: src/util/svgUtils/types.ts (hunk 1 above)
Comment (BuddhikaGeasman) [reply src/util/svgUtils/types.ts:2]: It is called the universal key. i kept it as it is, i wasn't use we use it. ps://github.com/wooorm/svg-element-attributes/blob/main/index.js#L7

File: src/util/svgUtils/types.ts (hunk 2 above)
Comment (BuddhikaGeasman) [reply src/util/svgUtils/types.ts:1]: Yes i would clean it, i was planning to do an update after i received some feedback!

File: src/util/svgUtils/types.ts (hunk 1 above)
Comment (ckifer) [reply src/util/svgUtils/types.ts:2]: we could probably omit this one

File: src/util/ReactUtils.ts (hunk 4 above)
Comment (BuddhikaGeasman) [reply src/util/ReactUtils.ts:220]: this was me, messing with things!

File: src/util/ReactUtils.ts (hunk 3 above)
Comment (BuddhikaGeasman) [reply src/util/ReactUtils.ts:217]: I created the `getSVGElementAttributesInCamelCase` cause the `svgElementAttributes` object structured as follows. I should have simply create the util to return the converted object without having to do the conversion multiple times. 
```js
circle: [
    'alignment-baseline',
    'baseline-shift',
    'clip',
    'clip-path',
    'clip-rule',
    'color',
    ...
    ]
    ```

Comment (ckifer) [reply src/util/ReactUtils.ts:217]: You can probably do the conversion and once and keep it stored as camel case, we don't need to do it at runtime

File: src/util/svgUtils/types.ts (hunk 1 above)
Comment (BuddhikaGeasman) [reply src/util/svgUtils/types.ts:2]: done

**Comment by ckifer** [comment]
Going to close this for now since follow ups haven't been addressed. Can re-open later

**Pull request closed without merging**